src/relations.jl

```diff
@@ -1154,33 +1154,20 @@ end
  - `Liquid()`, `Ice()` - liquid or ice phase to dispatch over.
  - `ts` thermodynamic state
 
-Returns supersaturation (qv/qv_sat -1) over water or ice.
+Returns supersaturation (pv/pv_sat -1) over water or ice.
 """
 function supersaturation(
     param_set::APS,
     q::PhasePartition{FT},
     ρ::FT,
     T::FT,
-    ::Liquid,
-) where {FT <: Real}
-
-    q_sat::FT = q_vap_saturation_generic(param_set, T, ρ, Liquid())
-    q_vap::FT = vapor_specific_humidity(q)
-
-    return q_vap / q_sat - FT(1)
-end
-function supersaturation(
-    param_set::APS,
-    q::PhasePartition{FT},
-    ρ::FT,
-    T::FT,
-    ::Ice,
+    phase::Phase,
 ) where {FT <: Real}
 
-    q_sat::FT = q_vap_saturation_generic(param_set, T, ρ, Ice())
-    q_vap::FT = vapor_specific_humidity(q)
+    p_v_sat::FT = saturation_vapor_pressure(param_set, T, phase)
```

Comment:
It's difficult for me to use the GH suggestion, but let's make `p_v_sat` an optional argument (since it's relatively expensive to compute):

```julia
    ...
    phase::Phase,
    p_v_sat::FT = saturation_vapor_pressure(param_set, T, phase)
) where {FT <: Real}

   ...
```

Reply:
So you want one version of `supersaturation` where `p_v_sat` is provided as input and another when it's computed from `T`?

Reply:
I think we can just move the `p_v_sat::FT = ...` up a few lines to make a single method with an optional argument.

Reply:
So we would no longer support `supersaturation(param_set, ts, phase)`. And instead support `supersaturation(p, p_sat)` ?

Reply:
The change I'm suggesting is non-breaking compared to the original PR, I just want `p_v_sat` to be an optional argument in `supersaturation`. `supersaturation(::ThermoParams, ::ThermoState)` can still call that method, and it doesn't need to pass the optional argument-- it's just helpful to have that argument optional in case we call this function from somewhere that `p_v_sat` has already been computed

Reply:
Gihub got really confused with the changes, it's better to just take a look at the file. 

I now have `supersaturation(..., phase)` and `supersaturation(..., p_v_sat)` The user facing method `supersaturation(params, ts, phase` goes back to `supersaturation(..., phase)`